client/landing/gutenboarding/components/domain-picker/suggestion-item.tsx

```diff
@@ -23,6 +27,8 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
 } ) => {
 	const { __ } = useI18n();
 
+	const [ , domainName, domainTld ] = suggestion.domain_name.match( rxDomainParts );
```

Comment:
This is resulting in type check error:
`Error TS2488: Type 'RegExpMatchArray | null' must have a '[Symbol.iterator]()' method that returns an iterator.`

Reply:
a type guard would solve this 🤔 

```suggestion
	const parsedResult: RegExpMatchArray | null = suggestion.domain_name.match( rxDomainParts );
	let domainName, domainTld;
	if ( parsedResult !== null ) {
		[ , domainName, domainTld ] = parsedResult;
	}
```

Reply:
My saviour! Now I see how it works!